### DISCUSSION TITLE
Failed deploy to dokku

### DISCUSSION THREAD

**Discussion by hoonpark7442** [dokku/dokku#4930] [Q&A] [answered]
sorry for my ugly english.
we have Ruby on Rails app and we need to setup staging server with dokku.
when I deployed, I got these failures:
```
git push looking-staging staging:master
      (...)
       Fetching multi_xml 0.6.0
       Installing multi_xml 0.6.0
       Fetching pagy 4.11.0
       Installing pagy 4.11.0
       Fetching pg 1.2.3
       Installing pg 1.2.3 with native extensions
       Fetching puma 3.12.6
       Installing puma 3.12.6 with native extensions
       Killed
       
       !
       !     Failed to install gems via Bundler.
       !
remote: 2021/11/16 08:18:48 exit status 1
To 13.124.49.7:looking-staging
 ! [remote rejected] staging -> master (pre-receive hook declined)
error: failed to push some refs to 'dokku@13.124.49.7:looking-staging'
```
it is stuck for about 2 hours installing puma gem and show errors at the end.
first i thought it was because of puma version, so i updated puma(3.12.6 -> 4.3.8) and ran again but i got these error too, and, this time, stuck at installing websocket-driver.
```
remote: + read -r line
remote: + [[ Installing websocket-driver 0.7.5 with native extensions == --* ]]
remote: + [[ Installing websocket-driver 0.7.5 with native extensions == ==* ]]
      ' 'Installing websocket-driver 0.7.5 with native extensions'
       Installing websocket-driver 0.7.5 with native extensions
remote: + read -r line

# it stucks for 2 hours...
```

is it because of each gem version? or memory issu? or any other reason for this failure? 

here is some info related to my app and ec2 instance
```
ruby '2.7.2'

gem 'rails', '~> 6.0.3.4'   # my rails version is Rails 6.0.3.7
gem 'puma', '~> 4.3' # puma version is 4.3.8
```

**my ec2 instance info**
Instance   | vCPU* | CPU Credits/hour | Mem (GiB) | Storage    | Network Performance (Gbps)***
t3a.micro |     2      |                12            |          1         | EBS-Only | Up to 5

**Comment by Cellane** [comment, marked as answer]
To my inexperienced eye, that looks like a memory issue, especially considering that the VM only has 1 GB RAM available, with likely no swap. I was experiencing similar issues when compiling an Elixir application on a 1 GB RAM Lightsail instance, and I solved (or worked around) the issue by mounting a 2 GB swap file.

**Reply by hoonpark7442** [reply]
thank you so much! i scaleup my ec2 instance and it works!
